diff --git a/libs/librepcb/core/project/bomgenerator.cpp b/libs/librepcb/core/project/bomgenerator.cpp
index 66d774aaa..c622a8907 100644
--- a/libs/librepcb/core/project/bomgenerator.cpp
+++ b/libs/librepcb/core/project/bomgenerator.cpp
@@ -61,9 +61,10 @@ std::shared_ptr<Bom> BomGenerator::generate(
   // Parse custom attributes.
   QStringList customCommonAttributes;
   QStringList customPartAttributes;
-  foreach (const QString& attribute, mAdditionalAttributes) {
+  foreach (QString attribute, mAdditionalAttributes) {
     if (attribute.endsWith("[]")) {
-      customPartAttributes.append(attribute.chopped(2));
+      attribute.chop(2);
+      customPartAttributes.append(attribute);
     } else {
       customCommonAttributes.append(attribute);
     }
